Tensile/Source/lib/include/Tensile/PropertyMatching.hpp

```diff
@@ -76,6 +76,10 @@ namespace Tensile
             virtual std::tuple<ReturnValue, double> findBestMatch(Object const& object,
                                                                   Transform transform) const = 0;
 
+            virtual ReturnValue findBestEvaluationSolution(Object const&   object,
+                                                           Hardware const& hardware,
+                                                           Transform       transform) const = 0;
```

Comment:
It almost makes more sense to me to create your own matching table class instead of piggy backing onto DistanceMatchingTable and forcing every other descendant to implement the new findBestEvaluationSolution interface. 

Your new implementation finds the best match based on TAM metrics, and does no 'distance matching' based on sizes. Thus I feel it makes more sense to upgrade the findBestMatch interface to include the hardware reference and implement a separate TAM matching table class that implements findBestMatch instead.

Of course this is my opinion and we can discuss more if you like.

Reply:
I am not sure how the final version of the TAM is going to be instrumented. 

What you are suggesting would be require the TAM to be recognized as part of de-serialization (for the findBestSolution interface dispatch the selection correctly) and that would require modification to the logic file. 

The requirement here is for a runtime switch and have it work using the existing logic.

At this time the feature is for evaluation. I don't expect this version will survive without revisions and if the TAM is proven the final version may be as you suggest here.